### PR TITLE
Mock home page for docs contributors

### PR DESCRIPTION
**Pull request by Integrayshaun**
## Why?
Volunteers who help us with our docs can no longer build the site locally because they don't have permission to query the storybookProjects API. `yarn start:doc-only` mocks that call to unblock them.

## What changed
- Add `CreateHomePage` function to mock or build the home page based on `process.env.GATSBY_DOCS_ONLY`

### CODE REVIEW

File: src/components/screens/IndexScreen/IndexScreen.js [hunk 1]
```diff
@@ -116,6 +121,7 @@ export function IndexScreen({
 
   return (
     <>
+      <GlobalStyle />
```
Comment (winkerVSbecks) [inline src/components/screens/IndexScreen/IndexScreen.js:124]: Why is this necessary?

File: src/pages/index.js [hunk 2]
```diff
@@ -1,97 +0,0 @@
...
-      <IndexScreen
-        latestBlogPost={latestPost}
-        projects={projects}
-        npmDownloads={npmDownloads}
-        twitterFollowerCount={twitterFollowerCount}
-        discordMemberCount={discordMemberCount}
-        githubContributorCount={githubContributorCount}
-        youTubeSubscriberCount={youTubeSubscriberCount}
-      />
-    </Fragment>
```
Comment (Integrayshaun) [inline src/pages/index.js:46]: @winkerVSbecks it was being added here in the template file that was deleted.

It's used in all other screen components and we don't get all of our style rules without it :(

File: src/components/screens/IndexScreen/IndexScreen.js (hunk 1 above)
Comment (Integrayshaun) [reply src/components/screens/IndexScreen/IndexScreen.js:124]: It's needed here now that the index template file has been deleted
https://github.com/storybookjs/frontpage/pull/428#discussion_r1033651834

**Review by winkerVSbecks — APPROVED** [review]
